src/librustc/ty/maps.rs

```diff
@@ -96,9 +96,9 @@ impl<'tcx> Value<'tcx> for Ty<'tcx> {
     }
 }
 
-pub struct CycleError<'a, 'tcx: 'a> {
+pub struct CycleError<'tcx> {
     span: Span,
-    cycle: RefMut<'a, [(Span, Query<'tcx>)]>,
+    cycle: Vec<(Span, Query<'tcx>)>,
```

Comment:
The reason for it was that, when we call `report_cycle`, we hold the borrow-mut lock, which then messes up my attempts to use `try_get` from within the pretty-printing code (it fails to acquire the `RefCell` lock).